radar_control/radar_control.py

```diff
@@ -186,70 +210,80 @@ def search_for_experiment(socket, status):
                 if isinstance(new_exp, ExperimentPrototype):
                     experiment = new_exp
                     new_experiment_received = True
-                    print("NEW EXPERIMENT FOUND")
+                    if __debug__:
+                        print("NEW EXPERIMENT FOUND")
                     break
                 else:
                     experiment = None
                     new_experiment_received = False
-                    print("NO NEW EXPERIMENT. CONTINUING.")
+                    if __debug__:
+                        print("RECEIVED AN EXPERIMENT NOT OF TYPE EXPERIMENT_PROTOTYPE. CANNOT RUN.")
+                    # TODO decide what to do here. I think we need this case if someone doesn't build their experiment
+                    # properly
                     break
         else:
-            print("No Experiment Provided - Continuing to Poll")
+            print("No Experiment Provided - Continuing to Poll") # TODO : check if this if socket in socks is necessary.
+                                                                 # Was based on an example, but should be tested.
 
     return new_experiment_received, experiment
 
 
 def verify_completed_sequence(tx_poller, tx_rx_poller, tx_socket, rx_ack_socket, rx_time_socket,
                               sigprocpacket, driverpacket, poll_timeout, seqnum, ):
-    """
```

Comment:
TODO: Based on Brian, we need to revisit this function, therefore we haven't looked through it.
maybe a tight loop that continually polls both sockets one after the other  with a very short timeout

Reply:
won't it just be one socket now? I figured this function could basically be moved (and modified as necessary) to Brian